### ISSUE TITLE
Best practice for using params in loaders

### ISSUE THREAD

**Issue by ebello** [mercurius-js/mercurius#362]
I'm wondering how best to treat params in a loader. For example, let's extend the example in the [loaders test](https://github.com/mercurius-js/mercurius/blob/master/test/loaders.js#L52) so that dogs can have multiple caretakers with different roles. We want to extend the owners query with a param to filter by role:
```
const query = `{
  dogs {
    name,
    caretakers(role: "trainer") {
      name
    }
  }
}`
```

In the loader function itself, I'm wondering how best to extract the role. It's my understand that each of the params must be the same for each nested query, is that right? Would it be safe to take the params from the first query as the role, or would you recommend matching the params by index? The following example would be simplest, I think. Or am I thinking about this the wrong way? Thank you.

```
const loaders = {
  Dog: {
    async caretakers(queries, { reply }) {
      const { role } = queries[0].params
      return queries.map(({ obj }) => {
        const allDogCaretakers = caretakers[obj.name])
        return allDogCaretakers.filter(caretaker => caretaker.role === role)
      }
    }
  }
}
```

**Comment by mcollina** [comment]
I don't think you can assume the role would be identical as you can include two GraphQL queries in the same payload with different roles.
